### DISCUSSION TITLE
Custom Exporter | Now able to integrate custom exporter writing in java-sdk with java-instrumentation

### DISCUSSION THREAD

**Discussion by NikunjPatel31** [open-telemetry/opentelemetry-java#6925] [Q&A]
I want to write all the traces into file before they are send to collector. For this I made necessary changes in `**OtlpHttpSpanExporter -> export**` method in Opentelemetry-java ( https://github.com/open-telemetry/opentelemetry-java ) . Below is the code of export method.

```
@Override
  public CompletableResultCode export(Collection<SpanData> spans) {
    String path = "/home/demo/testing.txt";
    try (BufferedWriter writer = Files.newBufferedWriter(Paths.get(path), UTF_8)) {
      writer.write("This is testing of opentelemetry");
      writer.flush();
    } catch (IOException e) {
      return marshaler.export(spans);
    }
    return marshaler.export(spans);
  }
```

Now I am running `gradle build -x test --warning-mode all` command to create .jar file. Once this jar file is generated I am add this jar to `project settings -> Libraries -> New Project Library` from intellij. Then I am generating .jar file of agent by running following command  `gradle build -x test --warning-mode all`. Once this .jar file is generated I am using this .jar file to bind it with my application.

Now whenever I am trying to generate trace, file is not getting created. I am able to see the traces at jaejar but file is not getting generated. 

I have tried to replace the .jar file from .m2/ directory but still the result is same. 

Please let me know what I am doing wrong.

**Comment by ashish159357** [comment]
facing same issue. don't know how to do changes in sdk and then use it in java-instrumentation

**Reply by trask** [reply]
If you're asking how to customize the SDK inside of the Java agent, check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions

**Comment by HarshMehta112** [comment]
I'm struggling with modifying the SDK and applying some changes in Java instrumentation. I'm not sure how to go about it.
